es6-shim.js

```diff
@@ -2013,7 +2017,7 @@
     // some environments don't have setTimeout - no way to shim here.
     if (typeof setTimeout !== 'function' && typeof setTimeout !== 'object') { return; }
 
-    ES.IsPromise = function (promise) {
+    var IsPromise = function (promise) {
```

Comment:
Because mutating an object after you've defined it ensures that `IsPromise` won't be optimized.  The number of slots is fixed at definition time.  But even if you add `IsPromise: null` to the definition of `ES` to hold the space, this ends up being initialized as a property reference stored in the ES object, not a "constant function reference" stored in the object's map (which can then be inlined after a hoisted type check).

This is described a bit by http://mrale.ph/blog/2012/09/23/grokking-v8-closures-for-fun.html